contracts/VoucherHub.sol

```diff
@@ -177,6 +177,25 @@ contract VoucherHub is
         emit VoucherCreated(voucherAddress, owner, voucherExpiration, voucherType, value);
     }
 
+    /**
+     * Top up a voucher by increasing its balance and extending the validity
+     * period before expiration.
+     * @param voucher The address of the voucher.
+     * @param value The amount of credits to top up.
+     */
+    function topUpVoucher(address voucher, uint256 value) external onlyRole(VOUCHER_MANAGER_ROLE) {
+        VoucherHubStorage storage $ = _getVoucherHubStorage();
+        require($._isVoucher[voucher], "VoucherHub: unknown voucher");
+        if (value > 0) {
```

Comment:
is there a case where value is not > 0 ?

Reply:
I expect request where user wants someone to extend its credits without recharging value
Do you (& @zguesmi too) want me to remove the `if (value > 0)` so worst case (from gas point of view) is triggering transfer with zero value?
Other point, Imo it might not be a good idea to forbid empty value, it looks best to let it open

Reply:
How about splitting into:
1. topUpVoucher(voucher, value) => fails if not value == 0
2. extandExpiration(voucher)